src/reducetensor.cpp

```diff
@@ -1031,6 +1032,10 @@ void ReduceTensorDescriptor::ReduceTensor(const Handle& handle,
                                        std::to_string(static_cast<int>(use_padding.first)) +
                                        std::to_string(static_cast<int>(use_padding.second));
 
+        int alignment            = 64;
+        Data_t workspace_aligned = std::align(
+            alignment, workspaceSizeInBytes - alignment, workspace, workspaceSizeInBytes);
```

Comment:
Actually we do not need `workspace_aligned`.
```suggestion
        if(nullptr == std::align(                                  //
            workspaceAlignRequirementBytes,                        //
            workspaceSizeInBytes - workspaceAlignRequirementBytes, //
            workspace,                                             //
            workspaceSizeInBytes))
        {
            MIOPEN_THROW(...);
        }
```
and revert all the changes below.

Reply:
[Informative] May be replaced by
```
workspace &= ~(workspaceAlignRequirementBytes - 1); // clear some LSBs of the pointer
workspaceSizeInBytes -= workspaceAlignRequirementBytes; // remove all spare space
```

Reply:
No, NO, NO definitely this is wrong.
`workspace &= ~(workspaceAlignRequirementBytes - 1);`  INCORRECT
The pointer value will be less than the value passed by the user. This way we will get a pointer to the area before or equal to the user ptr.
The aligned value should always be >= user value.

Reply:
[informative] @shurale-nkn Good catch! yeah it should be like
```
workspace += workspaceAlignRequirementBytes; // move ptr forwards first
workspace &= ~(workspaceAlignRequirementBytes - 1); // clear some LSBs of the ptr (moves ptr backwards)
workspaceSizeInBytes -= workspaceAlignRequirementBytes; // remove all spare space
```